pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/assignment/instance/InstanceTagPoolSelectorTest.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.controller.helix.core.assignment.instance;
+
+import static org.testng.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.helix.model.InstanceConfig;
+import org.apache.pinot.common.utils.config.InstanceUtils;
+import org.apache.pinot.spi.config.table.assignment.InstanceTagPoolConfig;
+import org.testng.annotations.Test;
+
+
+public class InstanceTagPoolSelectorTest {
+
+  private InstanceTagPoolConfig _tagPoolConfig;
+  private String _tableNameWithType = "tableNameWithType";;
+  private static final String _tag = "testTag";
+
+  @Test
+  public void testSelectInstances_PoolBasedTagConfig() {
```

Comment:
Add description on exactly what is being tested.

Also, use camelCase names, eg, `testPoolTagBasedInstanceSelection`.

Reply:
Done